### PR TITLE
fix: resolve AWS SSO/assumed-role profiles instead of silent fallback (#364)

### LINKED ISSUE #364: fix: AWS SSO/assumed-role profile ignored, silent fallback to default credentials
## Summary

When the selected AWS profile is an **SSO / assumed-role profile** (e.g. AWS IAM Identity Center), `envilder` silently ignores it and falls back to the **default** credential chain (static IAM user, env vars, etc.). Both `--profile <name>` and `$config.profile` in the map file are affected.

The command does **not** error — it succeeds against the *wrong* account, which makes the failure very hard to diagnose. Secrets get read from / pushed to the default account instead of the intended one.

## Environment

- envilder: `0.11.0`
- node: `v24.13.0`
- OS: Windows 11 (PowerShell)
- Provider: AWS
- Intended profile: SSO profile (`AWSReservedSSO_AdministratorAccess`, assumed-role) — account `A` (e.g. `387...`)
- Default profile: IAM user with static credentials — account `B` (e.g. `425...`)

## Reproduction

Map file `map.json`:

```json
{
    "$config": { "provider": "aws", "profile": "ssoProfile" },
    "CDK_VERSION": "/cdk-bootstrap/hnb659fds/version"
}
```

`/cdk-bootstrap/hnb659fds/version` exists **only** in account `A` (the SSO profile account), not in the default account `B`.

```bash
# AWS CLI honours the SSO profile correctly:
aws ssm get-parameter --name "/cdk-bootstrap/hnb659fds/version" --profile ssoProfile --query "Parameter.Value" --output text
# => 32   (read from account A)

# envilder, same SSO profile, via --profile flag:
envilder --map=map.json --envfile=.env --profile ssoProfile
# => Warning: No value found for: '/cdk-bootstrap/hnb659fds/version'

# envilder, same SSO profile, via $config.profile (no flag):
envilder --map=map.json --envfile=.env
# => Warning: No value found for: '/cdk-bootstrap/hnb659fds/version'
```

A secret that exists in **both** accounts is resolved fine, masking the bug — the value just silently comes from the default account.

The same happens on `--push`: with `$config.profile = ssoProfile`, the value was written to the **default** account `B`, not to `A`.

## Expected

- `envilder` resolves credentials from the requested SSO profile (matching AWS CLI / SDK v3 `fromNodeProviderChain({ profile })` behaviour, including SSO token resolution).
- If the profile cannot be resolved, it **errors loudly** instead of silently falling back to default credentials.

## Actual

- SSO / assumed-role profiles are ignored.
- Silent fallback to the default credential chain → reads/writes hit the wrong AWS account with no warning.

## Likely cause

The AWS credential provider is probably constructed without full SSO support (or the `profile` option is not threaded into `fromNodeProviderChain` / `fromSSO`). AWS SDK v3 needs `@aws-sdk/credential-providers` `fromNodeProviderChain({ profile })` (which includes `fromSSO`) for SSO profiles to work.

## Impact

- Secrets pushed/pulled against the wrong account without any error.
- Particularly dangerous for `--push` (writes secrets to an unintended account).

### PR DESCRIPTION
**Pull request by macalbert**
## Summary

Fixes #364 — AWS SSO / assumed-role profiles were silently ignored, falling
back to the default credential chain. This caused secrets to be read from /
pushed to the **wrong AWS account with no error**, affecting both
`--profile <name>` and `$config.profile`.

Adding a reproduction test per stack surfaced a genuine failure in **three** of
the four stacks — not just the CLI.

## Root cause per stack

| Stack | Root cause | Fix |
|-------|-----------|-----|
| CLI core | `fromIni({ profile })` does not resolve SSO profiles | `fromNodeProviderChain({ profile })` (includes `fromSSO`) |
| Node.js SDK | same `fromIni` limitation | `fromNodeProviderChain({ profile })` |
| .NET SDK | `CredentialProfileStoreChain` recognises the SSO profile but the runtime SSO assemblies were missing (`AWSSDK.SSOOIDC could not be loaded`) | add `AWSSDK.SSO` + `AWSSDK.SSOOIDC` package references |
| Python SDK | none — `boto3.Session(profile_name=...)` resolves SSO natively | regression guard test only |

## TDD cycle

- **🔴 RED** — one reproduction test per stack. CLI, Node and .NET failed;
  Python passed (documents existing correct behaviour).
- **🟢 GREEN** — minimal production fixes for CLI, Node and .NET.
- **♻️ REFACTOR** — consolidated a duplicated CLI-core test.

## Review fixes

- Normalize (`trim`) `config.profile` in the CLI core before truthiness check,
  consistent with the Node.js SDK (Gemini).
- Harden the CLI-core SSO test isolation: reset the credential mock and assert
  call count (Copilot).
- Exclude `github-action/dist/**` from the Biome pre-commit hook — it was
  reformatting the ncc bundle on commit and breaking the `verify-build` check.

## Release

Ships the fix as a patch across all affected packages:

- CLI `0.11.0` → `0.11.1`
- Node.js SDK `0.3.0` → `0.3.1`
- .NET SDK `0.4.0` → `0.4.1`

Changelogs updated (`[Unreleased]` → versioned, dated 2026-06-25).

## Branding

- Add the Envilder logo (`assets/logo.png`, 1024×1024) as the **NuGet package
  icon** (`<PackageIcon>`), verified embedded via `dotnet pack`.
- Add the Envilder banner to the top of the .NET, Node.js and Python SDK
  READMEs (npm/PyPI/NuGet.org render the README; no native icon field there).

## Validation

- Core / CLI (incl. e2e): **274 passed**
- Website i18n + changelog integrity: **39 passed**
- .NET `SecretProviderFactoryTests`: **12 passed** (incl. new SSO test)
- Node SDK factory: **8 passed** · Python factory: **68 passed**
- `pnpm lint`, `pnpm verify:gha`, `dotnet pack`: green

Closes #364

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed AWS SSO / IAM Identity Center and assumed-role profile credential resolution to use the full AWS credential provider chain instead of falling back to default credentials (preventing access to the wrong account).
  * Applied the fix across the supported SDKs’ profile-based AWS flows and ensured SSO resolution works at runtime.

* **Documentation**
  * Updated pull/push command docs and SDK/website readmes/changelog entries with active SSO session guidance.
  * Added AWS profile mapping documentation, including translations.

* **Tests**
  * Added/updated unit tests to verify profile-based AWS credential resolution via the SSO-capable credential chain.

### CODE REVIEW

File: src/envilder/core/infrastructure/aws/AwsSecretProviderFactory.ts [hunk 1]
```diff
@@ -8,7 +8,9 @@ export function createAwsSecretProvider(
   config: MapFileConfig,
 ): ISecretProvider {
   const ssm = config.profile
-    ? new SSM({ credentials: fromIni({ profile: config.profile }) })
+    ? new SSM({
+        credentials: fromNodeProviderChain({ profile: config.profile }),
+      })
     : new SSM();
```
[the remark this answers is not in the document]
Comment (macalbert) [reply src/envilder/core/infrastructure/aws/AwsSecretProviderFactory.ts]: **Resolved.** Trimmed `config.profile` before the truthiness check, so whitespace-only profiles fall back to the default credential chain — consistent with the Node.js SDK's `normalize()` helper.

**File:** `src/envilder/core/infrastructure/aws/AwsSecretProviderFactory.ts`
**Commit:** [`c106051`](https://github.com/macalbert/envilder/commit/c106051)
**Evidence:** `pnpm vitest run` (CLI core) — 2 passed · `pnpm verify:gha` — ✓

File: tests/envilder/core/infrastructure/aws/AwsSecretProviderFactory.test.ts [hunk 2]
```diff
@@ -8,36 +8,38 @@ vi.mock('@aws-sdk/client-ssm', () => {
...
-    expect(fromIni).toHaveBeenCalledWith({ profile: 'myprofile' });
+    expect(fromNodeProviderChain).not.toHaveBeenCalled();
   });
 
-  it('Should_ReturnAwsProvider_When_NoProfileProvided', () => {
+  it('Should_UseSsoCapableCredentialChain_When_ProfileProvided', () => {
     // Arrange
-    const config = { provider: 'aws' };
-    vi.mocked(fromIni).mockClear();
+    const config = { provider: 'aws', profile: 'myprofile' };
```
[the remark this answers is not in the document]
Comment (macalbert) [reply tests/envilder/core/infrastructure/aws/AwsSecretProviderFactory.test.ts:35]: **Resolved.** Added `vi.mocked(fromNodeProviderChain).mockClear()` in Arrange and an `toHaveBeenCalledTimes(1)` assertion, preventing cross-test leakage and strengthening the regression guard.

**File:** `tests/envilder/core/infrastructure/aws/AwsSecretProviderFactory.test.ts`
**Commit:** [`cbef4bc`](https://github.com/macalbert/envilder/commit/cbef4bc)
**Evidence:** `pnpm vitest run` (CLI core) — 2 passed

File: tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs [hunk 3]
```diff
@@ -214,6 +214,35 @@ public void Should_SelectAwsProvider_When_NoAwsRegionConfigured()
...
+		Directory.CreateDirectory(_tempDirToDelete);
+		var credentialsFile = Path.Combine(_tempDirToDelete, "credentials");
+		File.WriteAllText(credentialsFile, string.Join('\n',
+			"[sso-profile]",
+			"sso_start_url=https://example.awsapps.com/start",
+			"sso_region=us-east-1",
+			"sso_account_id=123456789012",
+			"sso_role_name=AdministratorAccess",
+			"region=us-east-1"));
+		OverrideEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", credentialsFile);
```
[the remark this answers is not in the document]
Comment (macalbert) [reply tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs]: **Resolved.** Hardened the test to be hermetic: it now overrides `AWS_CONFIG_FILE`, `AWS_PROFILE`, `USERPROFILE`, and `HOME` to the temp directory (mirroring `Should_SelectAwsProvider_When_NoAwsRegionConfigured`), so it can never read the host `~/.aws/config`.

Note on placement: the production path builds `new CredentialProfileStoreChain(AWS_SHARED_CREDENTIALS_FILE)` with an explicit location, which reads **only** that file — so writing the SSO profile to the temp credentials file is exactly the file production loads. The added isolation removes any chance of a fallback read.

**File:** `tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs`
**Commit:** [`e84ded5`](https://github.com/macalbert/envilder/commit/e84ded5)
**Evidence:** `dotnet test` (SSO test) — 1 passed

File: src/envilder/core/infrastructure/aws/AwsSecretProviderFactory.ts [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 
 export function createAwsSecretProvider(
   config: MapFileConfig,
 ): ISecretProvider {
-  const ssm = config.profile
-    ? new SSM({ credentials: fromIni({ profile: config.profile }) })
+  const profile = config.profile?.trim();
+  const ssm = profile
+    ? new SSM({ credentials: fromNodeProviderChain({ profile }) })
     : new SSM();
```
[the remark this answers is not in the document]
Comment (macalbert) [reply src/envilder/core/infrastructure/aws/AwsSecretProviderFactory.ts:13]: **Won't fix — by design.** Treating a blank/whitespace `profile` as "no profile" (→ default credential chain) is intentional and consistent with the Node.js SDK (`normalize()` maps empty/whitespace to `undefined`) and with how the AWS CLI treats an empty `--profile`. A whitespace-only value is configuration noise, not an explicitly chosen account.

The "wrong account" class of bug in #364 was the *opposite* failure: a **valid** SSO profile being ignored. That is now fixed — valid profiles resolve through `fromNodeProviderChain`. Erroring on whitespace would diverge from the Node SDK and add an edge case without protecting against the real issue.

Keeping the current behavior.

File: tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (macalbert) [reply tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs]: **Resolved.** This was largely addressed by the earlier hardening/refactor: `CreateIsolatedAwsHome()` already isolates `AWS_PROFILE`, `AWS_CONFIG_FILE`, `USERPROFILE`, and `HOME`. I've now also centralized `AWS_REGION` + `AWS_DEFAULT_REGION` nulling into that helper, so the SSO guard can never consult the host machine's AWS config.

**File:** `tests/sdks/dotnet/Infrastructure/SecretProviderFactoryTests.cs`
**Commit:** [`d0924e9`](https://github.com/macalbert/envilder/commit/d0924e9)
**Evidence:** `dotnet test` (SecretProviderFactoryTests) — 12 passed

File: tests/sdks/nodejs/infrastructure/secret-provider-factory.test.ts [hunk 5]
```diff
@@ -134,4 +135,18 @@ describe('SecretProviderFactory', () => {
...
+    const config = { provider: SecretProviderType.Aws, profile: 'sso-profile' };
+
+    // Act
+    createSecretProvider(config);
+
+    // Assert
+    expect(fromNodeProviderChain).toHaveBeenCalledWith({
+      profile: 'sso-profile',
+    });
+  });
```
[the remark this answers is not in the document]
Comment (macalbert) [reply tests/sdks/nodejs/infrastructure/secret-provider-factory.test.ts:152]: **Resolved.** Added `expect(fromNodeProviderChain).toHaveBeenCalledTimes(1)` so the guard now fails if the credentials chain is constructed more than once.

**File:** `tests/sdks/nodejs/infrastructure/secret-provider-factory.test.ts`
**Commit:** [`d0924e9`](https://github.com/macalbert/envilder/commit/d0924e9)
**Evidence:** `pnpm vitest run` (Node SDK factory) — 8 passed
